src/index.tsx

```diff
@@ -212,7 +212,8 @@ type NiceModalArgs<T> = T extends keyof JSX.IntrinsicElements | React.JSXElement
   ? Partial<Omit<React.ComponentProps<T>, 'id'>>
   : Record<string, unknown>;
 
-export function show<T extends any>(modal: React.FC<any>, args?: NiceModalArgs<React.FC<any>>): Promise<T>;
```

Comment:
@spaguette maybe it is possible to change it to 
`export function show<T extends any, P>(modal: React.FC<P>, args?: NiceModalArgs<React.FC<P>>): Promise<T>;
` - additional generic argument for modal props
I've tested it locally and didn't find any issues based on my codebase, maybe you know cases when it throws some error?

Reply:
AFAIU, it's possible by design of the library to call the `NiceModal.register` (see [here](https://github.com/eBay/nice-modal-react/pull/61/files#diff-0b5adbfe7b36e4ae2f479291e20152e33e940f7f265162d77f40f6bdb5da7405L397-L403)) globally with the `props` as the last argument.
This way, you can pass some "default" props I guess, and then override them with `show`, or just omit them 🤷 
Therefore I believe that just making it a generic `P` would not really work for this case, unfortunately, whereas `Partial<P>` would. However, the `Partial<P>` will miss the `null` / `undefined` checks that I would really like to have (though the flexible nature of the library might not allow it).